### DISCUSSION TITLE
GSOC 2024 Contributor

### DISCUSSION THREAD

**Discussion by man-myth** [unicode-org/icu4x#4680] [General]
Hi there,

I'm a fourth-year computer science student interested in working on the Auto-Generating ICU4X WASM project. I'm excited about the potential this project holds and am eager to contribute my skills and enthusiasm to its development. Looking forward to collaborating with you all!

**Comment by sffc** [comment]
Thanks for your interest. I am also excited about this project!

Do you have any questions?

**Reply by ayushkoli772** [reply]
Hello! Im also interested in this project as it perfectly aligns with my current skills. Are there any initial tasks you can assign me to demonstrate my capabilities and commitment to work on the project?

Thank you for your time and consideration.

**Reply by sffc** [reply]
A good starter project would be adding a new feature or component to the existing ICU4X WASM UI.

**Reply by ayushkoli772** [reply]
Thank you for your prompt response. Im currently working on setting dev environment and will start further task as soon as it is complete. Will it be fine if I reach out to you in case I face some issues?

Thank you again for your time and consideration!

**Reply by ayushkoli772** [reply]
Hello! @sffc I have successfully setup dev environment. Can you please guide me in running WASM demo website locally? Thank you :)

**Reply by sffc** [reply]
See tutorials/js.md for some very basic instructions. The WASM demo currently lives in tutorials/npm, and it is driven by npm scripts in package.json.

**Reply by ayushkoli772** [reply]
Hello! I gone through your given resources and also run on localhost. But there were errors like 

`ERROR in ./src/ts/app.ts 1:0-42
Module not found: Error: Can't resolve 'icu4x'`

for code- `import { ICU4XDataProvider } from 'icu4x';`

How to resolve this?

**Reply by ashu26jha** [reply]
Try the following:

```bash
npm install
npm run build
npm run start
```

**Comment by man-myth** [comment]
Hi, I've successfully set up the repository, and I'm currently reviewing the documentation and delving deeply into the project. Thank you!

**Comment by saumyabharti294** [comment]
Hello everyone,
Can i know if i can still contribute to "Expand Unicode Conformance Test Suite with More Data and Implementations". If the project is still active and has not assigned to anyone. As i have familiarity with more than 3 languages i.e C++, Rust, JavaScript,  and Java.

**Reply by sffc** [reply]
I encourage you to express your interest in that project on its repository, which can be found at https://github.com/unicode-org/conformance

**Reply by saumyabharti294** [reply]
Thank you, i have started to work on the project but have some doubts, Can you please tell me where should i reach for this?

**Reply by 0saurabh0** [reply]
@saumyabharti294 Here should be fine, also there is [gsoc@unicode.org](mailto:gsoc@unicode.org)

**Reply by sffc** [reply]
I encourage you to post questions and engage on https://github.com/unicode-org/conformance/discussions for the Unicode Conformance project.

**Comment by yatigautam** [comment]
hello everyone, I'm YATI GAUTAM, currently an undergraduate,I came across unicode yesterday, read about it and developed a good understanding of the projects listed. I think i might be a good fit for the work, i have a good practice in c++, java and git.
I will be starting off by solving some of the issues and will keep you informed throughout. 
Any advices or suggestions would be helpful.

**Comment by ankitmrj** [comment]
Greetings,
Respected Sir ,

I am first year undergraduate student, of Computer Science at State University from India. I am thrilled to work on project " Expand Unicode Conformance Test Suite with More Data and Implementations".

Please help me to learn and contribute more idea to this project.

How to get template for proposal ?.
Thanking You !

**Reply by sffc** [reply]
Please see https://github.com/unicode-org/conformance for the Unicode Conformance Test Suite project.

**Reply by sffc** [reply]
Hi @ankitmrj, this is just a reminder to make sure you submit your project proposal by April 2 at 18:00 UTC so that it can be considered. Thank you!

**Comment by 0saurabh0** [comment]
Hi, @sffc 
some questions regarding the "ML segmenter project":

- when writing tests, How are these metrics i.e. (accuracy, size, and speed) defined and measured, also if possible can you provide examples of test cases or scenarios that the test suite will cover?

- Are there any specific goals or benchmarks that the project aims to achieve in terms of model performance?

A brief overview would do the job.

**Reply by sffc** [reply]
Please see https://github.com/unicode-org/lstm_word_segmentation for the ML segmenter project.

**Comment by yatigautam** [comment]
hello,
Where could I find the article related to building the project on my system?

**Reply by sffc** [reply]
Please see CONTRIBUTING.md.

**Comment by elfkuzco** [comment]
Hi @sffc , I am interested in working on "Linux command line (sh/Perl/Python) interface to Google groups" and have an understanding of the project requirements. Looking at the Google API for groups, I can develop a Python command line that will take inputs either through a file or directly on the command line. However, it is mentioned that " _it would be much better if we could keep many/most groups constantly in sync by using programs to do automatic updates based on daily LGL reports about our constituents._". I would like to enquire the format of the LGL reports and any further guides on how to begin.

**Reply by rsnj5** [reply]
Hello, I am also interested in working on this project as I have already worked on Python and Bash scripting along with Google APIs. I am currently looking on the documentation of [LGL mailing report](https://help.littlegreenlight.com/category/112-mailings-templates-acknowledgments). Although this doesn't have the exact format of the report Unicode uses (there is option of customizing reports), I think it will be more or less the same.  As far as I understand, there will be members and other information in LGL reports that we need to parse and update the Google Groups accordingly.

**Reply by sffc** [reply]
Hi @elfkuzco and @rsnj5! I copied you and Rick McGowan on #4727 and we can discuss the Google Groups project there.
